mne/epochs.py

```diff
@@ -183,6 +184,93 @@ def _save_split(epochs, fname, part_idx, n_parts, fmt):
+
+    elif event_repeated == 'drop':
+        logger.info('{} Keeping the first occurrence and dropping all others.'
+                    .format(msg))
+        new_events = events[u_idxs]
+
+    elif event_repeated == 'merge':
+        logger.info('{} Creating new event code to reflect simultaneous '
+                    'events and updating event_id.'.format(msg))
+        new_events = events.copy()
+        non_u_evs = u_evs[counts > 1]
+        for ev in non_u_evs:
+
+            # indices at which the non-unique events happened
+            idxs = (events[:, 0] == ev).nonzero()[0]
+
+            # Figure out event_codes prior to events to be merged, fall back
+            # to 0, if heterogeneous codes exist
+            prior_codes = events[idxs, 1]
+            new_prior = 0
+            if len(np.unique(prior_codes)) == 1:
+                new_prior = np.unique(prior_codes)[0]
+
+            # Make an event_id for the merged event
+            ev_codes = events[idxs, 2]
+
+            if len(np.unique(ev_codes)) > 1:
+
+                new_event_id_key = list()
+                for code in ev_codes:
+                    # inverse dict lookup, because event_id is a one-to-one map
+                    kk = list(event_id.keys())[list(event_id.values())
+                                               .index(code)]
+                    new_event_id_key.append(kk)
+                new_event_id_key = '/'.join(new_event_id_key)
+
+                # Check if we already have a corresponding code
+                new_event_code = event_id.get(new_event_id_key, False)
+
+                # Else, make one and add it to the event_id dict
+                if not new_event_code:
+                    ev_codes = np.concatenate((np.array(list(event_id.values())),  # noqa: E501
+                                               events[:, 1:].flatten()),
+                                              axis=0)
+                    new_event_code = np.setdiff1d(np.arange(1, 9999999),
+                                                  ev_codes).min()
+                    event_id[new_event_id_key] = int(new_event_code)
+
+            # However, if duplicate time samples have same event code, revert
+            # to "drop" behavior
+            else:
+                new_event_code = ev_codes[0]
+
+            # Replace duplicate event times with merged event
+            new_events[idxs[0], 1] = new_prior
+            new_events[idxs[0], 2] = new_event_code
+            new_events = np.delete(new_events, idxs[1:], 0)
+
+    else:
+        raise ValueError('`event_repeated` must be one of "error", "drop",'
+                         ' "merge" but is "{}"'.format(event_repeated))
+
+    # Remove obsolete kv-pairs from event_id after handling
+    to_delete = list()
+    for key, val in event_id.items():
+        if val not in new_events[:, 1:].flatten():
+            to_delete.append(key)
+    for key in to_delete:
+        event_id.pop(key)
```

Comment:
```suggestion
    keys = new_events[:, 1:].flatten()
    event_id = {k: v for k, v in event_id.items() if v not in keys}
```

Reply:
ahh cool thanks @jona-sassenhagen - but you probably mean:

```suggestion
        keys = new_events[:, 1:].flatten()
        event_id = {k: v for k, v in event_id.items() if v in keys}
```

... because we only want to have kv-pairs in `event_id` that actually map to something in `events` array